classes/Currency.php

```diff
@@ -863,8 +863,12 @@ public static function getIdByIsoCode($isoCode, $idShop = 0, $forceRefreshCache
             $query = Currency::getIdByQuery($idShop, $includeDeleted);
             $query->where('iso_code = \'' . pSQL($isoCode) . '\'');
 
-            $result = (int) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue($query->build());
-            Cache::store($cacheId, $result);
+            $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue($query->build());
+            if (empty($result)) {
+                return 0;
+            }
+
+            Cache::store($cacheId, (int) $result);
```

Comment:
Better cast the return instead of the cached value (or both) Or else the first call will return a string and the next ones an int

Reply:
@jolelievre Fixed